region-cloner/scripts/scripts.lua

```diff
@@ -352,20 +439,29 @@ function run_job(job)
         if (job.clear_resource_entities) then
             table.insert(forces_to_clear_paste_area, "neutral")
         end
-        for x=1, job.times_to_paste do
-            clear_paste_area(job.tiles_to_paste_x, job.tiles_to_paste_y, x, job.bounding_box, forces_to_clear_paste_area, job.surface, job.entity_pool)
-            validate_entity_pool(job.high_priority_pool)
-            validate_entity_pool(job.moving_rolling_stock_pool)
-            validate_entity_pool(job.entity_pool)
-            validate_entity_pool(job.lite_entity_pool)
-            -- High prio
-            copy_entity_pool(job.player, job.high_priority_pool, {x = job.tiles_to_paste_x * x, y = job.tiles_to_paste_y * x}, job.surface, job.force)
-            -- rolling stock
-            copy_lite_entity_pool(job.player, job.moving_rolling_stock_pool, {x = job.tiles_to_paste_x * x, y = job.tiles_to_paste_y * x}, job.surface, job.force)
-            -- power poles
-            copy_lite_entity_pool(job.player, job.lite_entity_pool, {x = job.tiles_to_paste_x * x, y = job.tiles_to_paste_y * x}, job.surface, job.force)
-            -- rest
-            copy_entity_pool(job.player, job.entity_pool, {x = job.tiles_to_paste_x * x, y = job.tiles_to_paste_y * x}, job.surface, job.force)
-        end
+		
+		region_cloner_xmax = job.times_to_paste
+		region_cloner_job = job
+		region_cloner_drum = 0
+		progress_bar = mod_gui.get_frame_flow(region_cloner_job.player)[GUI_ELEMENT_PREFIX .. "advanced_view_pane"][GUI_ELEMENT_PREFIX .. "progress_bar"].state
+		detailed_log = mod_gui.get_frame_flow(region_cloner_job.player)[GUI_ELEMENT_PREFIX .. "advanced_view_pane"][GUI_ELEMENT_PREFIX .. "detailed_log"].state
+		if progress_bar then
+			--[With a progress bar]
+			region_cloner_x = 0
+			local text = string.format("Copied %d out of %d", region_cloner_x, region_cloner_xmax);	
+			game.print( text, {0,1,1})
+			
+			if detailed_log then
+				script.on_nth_tick(1, region_cloner_drum_machine)
+			else
+				script.on_nth_tick(1, region_cloner_for)   
+			end
+		else
```

Comment:
Bit weird to have detailed_log be selectable while progress is not. I think I'll change it to a dropdown.

Reply:
Or just let it work with only print detailed progress selected